eiffellib/publishers/rabbitmq_publisher.py

```diff
@@ -141,6 +150,16 @@ def _confirm_delivery(self, method_frame):
     def send_event(self, event, block=True):
         """Validate and send an eiffel event to the rabbitmq server.
 
+        This method will set the source on all events if there is a source
+        added to the :obj:`RabbitMQPublisher`.
+        If the routing key is set to None in the :obj:`RabbitMQPublisher` this
+        method will use the routing key from the event that is being sent.
+        The event domainId will also be added to `meta.source` if it is set to
+        anything other than the default value. If there is no domainId
+        set on the event, then the domainId from the source in the
+        :obj:`RabbitMQPublisher` will be used in the routing key, with a default
+        value taken from the :obj:`eiffellib.events.eiffel_base_event.EiffelBaseEvent`.
```

Comment:
Please help me with this documentation. I am not sure if this is clear enough.

Reply:
I think the description is clear, but the interface is pretty complicated. Maybe we're giving people too much rope with a source attribute in the publisher, but I'm sure it's convenient since the source is typically the same.

Reply:
We use that part heavily in ETOS, so it would have a big impact to remove it.
We could of course deprecate it and rework ETOS to add it to all events ETOS publishes, but I kind of like having it here.